WordPress/src/main/java/org/wordpress/android/GCMMessageService.java

```diff
@@ -122,30 +124,19 @@ private void handleDefaultPush(String from, @NonNull Bundle data) {
         EventBus.getDefault().post(new NotificationEvents.NotificationsChanged());
     }
 
-    private void trySaveNoteIfNotAlreadyInBucket(Bundle data) {
-        if (SimperiumUtils.getNotesBucket() != null) {
-            if (data == null) {
+    private boolean buildNoteObjectFromPNPayloadAndSaveIt(Bundle data) {
+           if (data == null) {
                 AppLog.e(T.NOTIFS, "Bundle is null! Cannot read '" + PUSH_ARG_NOTE_ID +"'.");
-                return;
+                return false;
             }
 
             String noteId = data.getString(PUSH_ARG_NOTE_ID, "");
-            try {
-                SimperiumUtils.getNotesBucket().get(noteId);
-                // all good if we got here
-            } catch (BucketObjectMissingException e) {
-                AppLog.e(T.NOTIFS, e);
-                SimperiumUtils.trackBucketObjectMissingWarning(e.getMessage(), noteId);
-
-                if (data.containsKey(PUSH_ARG_NOTE_FULL_DATA)) {
-                    //if note doesn't exist, try taking it from the PN payload, build it and save it
-                    // Simperium will take care of syncing local and server versions up at a later point
-                    String base64FullData = data.getString(PUSH_ARG_NOTE_FULL_DATA);
-                    Note note = new Note.Schema().buildFromBase64EncodedData(noteId, base64FullData);
-                    SimperiumUtils.saveNote(note);
-                }
+            String base64FullData = data.getString(PUSH_ARG_NOTE_FULL_DATA);
+            Note note = Note.buildFromBase64EncodedData(noteId, base64FullData);
+            if (note != null) {
+                return NotificationsTable.saveNote(note, true);
             }
-        }
+        return false;
```

Comment:
before returning false, I'd write to `AppLog` indicating that we're going to fail in accomplishing our mission (that is, the mission described by the method name `buildNoteObjectFromPNPayloadAndSaveIt`)